### PR TITLE
[Core] Adds tandem blow/strike traits to sql and mods to cpp/lua

### PR DESCRIPTION
**Pull request by MowFord**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Implements the core plumbing for bst job traits [Tandem Strike](https://www.bg-wiki.com/ffxi/Tandem_Strike) and [Tandem Blow](https://www.bg-wiki.com/ffxi/Tandem_Blow)

I have a few ideas on how to implement the actual bonuses, but I think it better to put this in first to avoid merge conflicts

## Steps to test these changes

Test that at the appropriate job levels you get the appropriate traits and mods

### CODE REVIEW

File: scripts/enum/mod.lua [hunk 1]
```diff
@@ -410,6 +410,10 @@ xi.mod =
     DARK_MAGIC_DURATION             = 1072, -- Increases Dark Magic spell durations by percentage (e.g. mod value 10 = +10% duration)
     ENHANCES_DARK_SEAL              = 1073, -- Enhances "Dark Seal" effect (Increases Dark Magic spell durations by 10% per Dark Seal merit while Dark Seal active)
 
+    -- Beastmaster
+    TANDEM_STRIKE                   = 271,  -- Grants a bonus to your and your pet's accuracy and magic accuracy when you and your pet are attacking the same target.
```
Comment (Xaver-DaRed) [inline scripts/enum/mod.lua]: Considering this is a modifier and not the trait itself, can we name this `TANDEM_STRIKE_BONUS`?

File: scripts/enum/mod.lua [hunk 2]
```diff
@@ -410,6 +410,10 @@ xi.mod =
...
+    TANDEM_BLOW                     = 272,  -- Reduces amount of TP gained by enemies when striking them if you and your pet are attacking the same target.
```
Comment (Xaver-DaRed) [inline scripts/enum/mod.lua]: Following that logic, `TANDEM_BLOW_BONUS` would apply here.

And of course, same with the ones in modifier.h

File: scripts/enum/mod.lua (hunk 1 above)
Comment (TeoTwawki) [reply scripts/enum/mod.lua]: `TANDEM_STRIKE_ACCURACY` fits better I think, bonus is usually a loaded word for things adding onto when we're naming mods

Comment (TeoTwawki) [reply scripts/enum/mod.lua:414]: except ya know, spell better than teo..

Comment (TeoTwawki) [reply scripts/enum/mod.lua]: likewise `TANDEM_BLOW_TP_REDUCTION` even if a bit wide.

Comment (Xaver-DaRed) [reply scripts/enum/mod.lua]: Well, considering its a bonus to both ACC and MACC, i thought a more generic name would fit better. Not opposd to it though

Comment (Xaver-DaRed) [reply scripts/enum/mod.lua]: ok then, lets go with `TANDEM_STRIKE_ACCURACY` and `TANDEM_BLOW_TP_REDUCTION`

Comment (MowFord) [reply scripts/enum/mod.lua]: good point about it not being the real mod, i went with `_POWER` to indicate we're just defining how strong it is

Comment (TeoTwawki) [reply scripts/enum/mod.lua]: `_POWER` sounds alright but might get changed somewhere down the road depending on future bs SE adds

**Review by Xaver-DaRed — APPROVED** [review]

File: scripts/enum/mod.lua (hunk 1 above)
Comment (TeoTwawki) [reply scripts/enum/mod.lua]: (if `_ACCMACC` or even `_ACC` happened I'd be gettin muh paddle)

File: documentation/player_traits.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+132	con anima
+133	con brio
+121	inspiration
+122	sleight of sword
+124	superior
+125	magic acc bonus
+126	magic eva bonus
+104	true shot
+135	max hp boost II
+136	shield barrier
```
Comment (TeoTwawki) [inline documentation/player_traits.txt]: Offset by 1536 from ROM\181\72.dat (so 1537 = 1)

[FFXI full trait list as of January 11th 2024.csv](https://github.com/LandSandBoat/server/files/13907756/FFXI.full.trait.list.as.of.January.11th.2024.csv)

Comment (TeoTwawki) [reply documentation/player_traits.txt]: I've been auto generating our enum for years btw. Just haven't kept up recently.

Comment (MowFord) [reply documentation/player_traits.txt]: omg so i didn't have to guess and check

Comment (MowFord) [reply documentation/player_traits.txt]: image

wow literally the last two traits eh

Comment (MowFord) [reply documentation/player_traits.txt]: added to the documentation txt in case people want to implement other missing traits

**Review by TeoTwawki — DISMISSED** [review]

**Comment by Xaver-DaRed** [comment]
So Teo, this trait adds a conditional value. Pretty much a "latent effect: same target"

Only when both master and pet are targeting the same entity. With separate modifiers, its easier to jus go to tp calculations, fetch actor and pet, check for same target and then just add the mod to tp gained/not gained

if we reused subtle blow II, the value would always be added... or removed... you know what i mean

**Comment by TeoTwawki** [comment]
> So Teo, this trait adds a conditional value. Only when both master and pet are targeting the same mob. With separate modifiers, its easier to jus go to tp calculations, fetch actor and pet, check for same target and then just add the mod to tp gained/not gained
> 
> if we reused subtle blow II, the value would always be added... or removed... you know what i mean

I was thinking we'd be adding mods only after it becomes active in some core logic - it'd kind of hard to guess how we'd use this when its latterly a modifier we aren't even checking for (I am glad the PR didn't use the word "implements"). As I can't see how this is intended to play out, I've recused myself from review

**Comment by Xaver-DaRed** [comment]
well, we can add a simple check to tp.lua in
`xi.combat.tp.calculateTPGainOnPhysicalDamage` and
`xi.combat.tp.calculateTPGainOnMagicalDamage`

```lua
local subtleBlowII = actor:getMod(xi.mod.SUBTLE_BLOW_II) -- This already exists
local pet          = actor:getPet()

if pet and actor:getTarget() == pet:getTarget() then
    subtleBlowII = subtleBlowII + actor:getMod(THE_NEW_MOD)
end
```
Everything else would be the same, I think

**Comment by MowFord** [comment]
Correct this isn't the implementation PR, just getting the traits and power mods implemented. My original thought was combat tick listeners on the pet to give/remove mods but I'm undecided on an implementation method just yet

**Comment by UmeboshiXI** [comment]
ASB has an implementation for [Tandem Strike](https://github.com/AirSkyBoat/AirSkyBoat/pull/3134/files) if you want to reference it for ideas on how to implement/not implement.

Edit: Tandem Blow to Tandem Strike

**Comment by MowFord** [comment]
Not interested in continuing this to the second stage of implementation. If someone wants to take this over after it's merged great, otherwise might be best to just close the PR

**Comment by Xaver-DaRed** [comment]
There is no need to close this PR nor any need to do any changes.
This PR implements certain missing job traits to the job trait list, adds modifiers for the job traits and adds documentation.
Thats it. Thats the scope of the PR. This PR is not intended to make this traits actuall work, and thats fne.

**Review by WinterSolstice8 — APPROVED** [review]

**Comment by TeoTwawki** [comment]
others have stamped, so I am hittin da button. Not really a fan of placeholdering mods that are unused, but I've done it before myself to evade rebase hell.

**Pull request merged**